pgmpy/factors/hybrid/FunctionalCPD.py

```diff
@@ -0,0 +1,122 @@
+
+        >>> from pgmpy.factors.hybrid import FunctionalCPD
+        >>> cpd = FunctionalCPD(
+        ...    variable="x3",
+        ...    fn=lambda parent_sample: np.random.normal(
+        ...        0.2 * parent_sample["x1"] + 0.3 * parent_sample["x2"] + 1.0, 1),
+        ...    parents=["x1", "x2"])
+
+        >>> cpd.variable
+        'x3'
+
+        >>> cpd.parents
+        ['x1', 'x2']
+        """
+        self.variable = variable
+        if not callable(fn):
+            raise ValueError("`fn` must be a callable function.")
+        self.fn = fn
+        self.parents = parents if parents else []
+        self.variables = [variable] + self.parents
+
+    def sample(self, n_samples=100, parent_sample=None):
+        """
+        Simulates a value for the variable based on its CPD.
+
+        Parameters:
+        ----------
+
+        n_samples: int, (default: 100)
+            The number of samples to generate.
+
+        parent_sample: pandas.DataFrame, optional
+            A DataFrame where each column represents a parent variable and rows are samples.
+
+        Returns:
+        -------
+        sampled_values: numpy.ndarray
+            Array of sampled values for the variable.
+
+        Examples
+        --------
+        >>> from pgmpy.factors.hybrid import FunctionalCPD
+        >>> cpd = FunctionalCPD(
+        ...    variable="x3",
+        ...    fn=lambda parent_sample: np.random.normal(
+        ...        1.0 + 0.2 * parent_sample["x1"] + 0.3 * parent_sample["x2"], 1),
+        ...    parents=["x1", "x2"])
+
+        >>> parent_samples = pd.DataFrame({'x1' : [5, 10], 'x2' : [1, -1]})
+        >>> cpd.sample(2, parent_samples)
+
+        """
+        if parent_sample is not None:
+            if not isinstance(parent_sample, pd.DataFrame):
+                raise TypeError("`parent_sample` must be a pandas DataFrame.")
+
+            if not all(parent in parent_sample.columns for parent in self.parents):
+                missing_parents = [
+                    p for p in self.parents if p not in parent_sample.columns
+                ]
+                raise ValueError(
+                    f"Missing values for parent variables: {missing_parents}"
+                )
+            if len(parent_sample) != n_samples:
+                raise ValueError("Length of `parent_sample` must match `n_samples`.")
+
+            sampled_values = []
+            for _, row in parent_sample.iterrows():
+                sampled_values.append(self.fn(row))
+
+            sampled_values = np.array(sampled_values)
+        else:
+            sampled_values = []
+            for _ in range(n_samples):
+                sampled_values.append(self.fn(parent_sample))
+
+            sampled_values = np.array(sampled_values)
+
+        return sampled_values
+
+    def __str__(self):
+        fn_name = (
+            "lambda(mean, std)" if self.fn.__name__ == "<lambda>" else self.fn.__name__
```

Comment:
When the function is lambda, let's just leave it to `"lambda fun."`. `mean` and `std` don't add any information and might not be relevant for some distributions.